shared/AppInsightsCore/src/JavaScriptSDK/EnvUtils.ts

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+"use strict";
+
+/**
+ * This file exists to hold environment utilities that are requied to check and
+ * validate the current operating environment. Unless otherwise required, please
+ * only defined methods (functions) in this class so that users of these 
+ * functions/properties only need to include those that are used within their own modules.
+ */
+
+export const strObject = "object";
+
+/**
+ * Checks if window object is available, this is required as we support the API running without a 
+ * window /document (eg. Node server, electron webworkers) and if we attempt to assign a window 
+ * object to a local variable or pass as an argument an "Uncaught ReferenceError: window is not defined" 
+ * exception will be thrown.
+ * Defined as a function to support lazy / late binding environments.
+ */
+export function hasWindow(): boolean {
+    return Boolean(typeof window === strObject && window);
+}
+
+/**
+ * Returns the global window object if it is present otherwise null.
+ * This helper is used to access the window object without causing an exception
+ * "Uncaught ReferenceError: window is not defined"
+ */
+export function getWindow(): Window|null {
```

Comment:
nit: `Window | null`, same for rest in this file

Reply:
I seem to have this for all of the items that return an object?

Reply:
Ah -- spacing...